src/Azure.Functions.ArtifactAssembler/CliArtifactZipper.cs

```diff
@@ -0,0 +1,78 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO.Compression;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Azure.Functions.ArtifactAssembler
+{
+    internal class CliArtifactZipper
+    {
+        private readonly string _rootWorkingDirectory;
+        public CliArtifactZipper(string rootWorkingDirectory)
+        {
+            _rootWorkingDirectory = rootWorkingDirectory;
+
+        }
+        internal void ZipCliArtifacts()
+        {
+            string stagingDirectory = Path.Combine(_rootWorkingDirectory, Constants.StagingDirName, Constants.CliOutputArtifactDirectoryName);
+
+            // Get all directories in the staging directory
+            var directories = Directory.GetDirectories(stagingDirectory);
```

Comment:
`Directory.EnumerateDirectories` will be helpful to avoid memory issues. (`Directory.GetDirectories` will load everything to memory)